src/Nethermind/Nethermind.Cli/Program.cs

```diff
@@ -30,45 +33,67 @@ namespace Nethermind.Cli
+        {
+            CommandLineApplication app = new() { Name = "Nethermind.Cli" };
+            _ = app.HelpOption("-?|-h|--help");
 
-        private static readonly CliModuleLoader ModuleLoader = new CliModuleLoader(Engine, NodeManager, CliConsole);
+            var colorSchemeOption = app.Option("-cs|--colorScheme <colorScheme>", "Color Scheme. Possible values: Basic|Dracula", CommandOptionType.SingleValue);
 
-        public static void Main()
-        {
-            RegisterConverters();
-            Engine.JintEngine.SetValue("serialize", new Action<JsValue>(v =>
+            app.OnExecute(() =>
+            {
+                ColorScheme? cs;
+                ICliConsole cliConsole = colorSchemeOption.HasValue() && (cs = MapColorScheme(colorSchemeOption.Value())) != null
+                    ? new ColorfulCliConsole(cs)
+                    : new CliConsole();
+
+                var historyManager = new StatementHistoryManager(cliConsole);
+                ILogManager logManager = new OneLoggerLogManager(new CliLogger(cliConsole));
+                ICliEngine engine = new CliEngine(cliConsole);
+                INodeManager nodeManager = new NodeManager(engine, Serializer, cliConsole, logManager);
+                var moduleLoader = new CliModuleLoader(engine, nodeManager, cliConsole);
+
+                RegisterConverters();
+                engine.JintEngine.SetValue("serialize", new Action<JsValue>(v =>
+                {
+                    string text = Serializer.Serialize(v.ToObject(), true);
+                    cliConsole.WriteGood(text);
+                }));
+
+                moduleLoader.DiscoverAndLoadModules();
+                ReadLine.AutoCompletionHandler = new AutoCompletionHandler(moduleLoader);
+
+                nodeManager.SwitchUri(new Uri("http://localhost:8545"));
+                historyManager.Init();
+                TestConnection(nodeManager, engine, cliConsole);
+                cliConsole.WriteLine();
+                RunEvalLoop(engine, historyManager, cliConsole);
+
+                cliConsole.ResetColor();
+                return ExitCodes.Ok;
+            });
+
+            try
             {
-                string text = Serializer.Serialize(v.ToObject(), true);
-                CliConsole.WriteGood(text);
-            }));
-
-            ModuleLoader.DiscoverAndLoadModules();
-            ReadLine.AutoCompletionHandler = new AutoCompletionHandler(ModuleLoader);
-
-            NodeManager.SwitchUri(new Uri("http://localhost:8545"));
-            HistoryManager.Init();
-            TestConnection();
-            CliConsole.WriteLine();
-            RunEvalLoop();
+                app.Execute(args);
+            }
+            catch (CommandParsingException)
+            {
+                app.ShowHelp();
+            }
         }
 
-        private static void TestConnection()
+        private static void TestConnection(INodeManager nodeManager, ICliEngine cliEngine, ICliConsole cliConsole)
```

Comment:
NP: please refrain from refactoring at the same time. Side note, I'd prefer previous style in this case.

Reply:
Not sure what you mean in this instance - is it the use od static fields instead of method params? If so, I feel this is a part of the change as the `cliConsole` and items which depend on it are now derived during runtime.

Reply:
You can also change just the static `CliConsole` and that would require less change. Of course, static var here is also not very nice.  In any case, its not a breaking change.